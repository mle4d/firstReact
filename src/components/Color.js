import React from 'react';

export default function Color() {
  const color = {
    name: 'Pretty Pink',
    rgb: '246 153 224',
    hex: '#F699E0'
  };
  const fields = Object.keys(color)
    .map(key => {
      return (
        <>
          <dt>{key}</dt>
          <dd style={{ color: '#F699E0' }}>{color[key]}</dd>
        </>
      );
    });
  return (
    <dl>
      {fields}
    </dl>
  );
}
