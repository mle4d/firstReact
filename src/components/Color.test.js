import React from 'react';
import { shallow } from 'enzyme';
import Color from './Color';

describe('App component', () => {
  it('renders App', () => {
    const wrapper = shallow(<Color />);
    expect(wrapper).toMatchSnapshot();
  });
});
