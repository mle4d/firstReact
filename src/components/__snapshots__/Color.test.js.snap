// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<dl>
  <dt>
    name
  </dt>
  <dd
    style={
      Object {
        "color": "#F699E0",
      }
    }
  >
    Pretty Pink
  </dd>
  <dt>
    rgb
  </dt>
  <dd
    style={
      Object {
        "color": "#F699E0",
      }
    }
  >
    246 153 224
  </dd>
  <dt>
    hex
  </dt>
  <dd
    style={
      Object {
        "color": "#F699E0",
      }
    }
  >
    #F699E0
  </dd>
</dl>
`;
